airbyte-integrations/connectors/destination-oracle/src/test-integration/java/io/airbyte/integrations/destination/oracle/OracleIntegrationTest.java

```diff
@@ -0,0 +1,194 @@
+    return retrieveRecordsFromTable(tableName, namespace);
+  }
+
+  @Override
+  protected List<String> resolveIdentifier(String identifier) {
+    final List<String> result = new ArrayList<>();
+    final String resolved = namingResolver.getIdentifier(identifier);
+    result.add(identifier);
+    result.add(resolved);
+    if (!resolved.startsWith("\"")) {
+      result.add(resolved.toLowerCase());
+      result.add(resolved.toUpperCase());
+    }
+    return result;
+  }
+
+  private List<JsonNode> retrieveRecordsFromTable(String tableName, String schemaName) throws SQLException {
+    List<org.jooq.Record> result = Databases.createOracleDatabase(db.getUsername(), db.getPassword(), db.getJdbcUrl())
+        .query(ctx -> ctx
+            .fetch(String.format("SELECT * FROM %s.%s ORDER BY %s ASC", schemaName, tableName, JavaBaseConstants.COLUMN_NAME_EMITTED_AT.substring(1)))
+            .stream()
+            .collect(Collectors.toList()));
+    return result
+        .stream()
+        .map(r -> r.formatJSON(JSON_FORMAT))
+        .map(Jsons::deserialize)
+        .collect(Collectors.toList());
+  }
+
+  private static Database getDatabase(JsonNode config) {
+    // todo (cgardens) - rework this abstraction so that we do not have to pass a null into the
+    // constru
+    // ctor. at least explicitly handle it, even if the impl doesn't change.
+    return Databases.createDatabase(
+        config.get("username").asText(),
+        config.get("password").asText(),
+        String.format("jdbc:oracle:thin:@//%s:%s/%s",
+            config.get("host").asText(),
+            config.get("port").asText(),
+            config.get("sid").asText()),
+        "oracle.jdbc.driver.OracleDriver",
+        null);
+  }
+
+  // how to interact with the mssql test container manually.
+  // 1. exec into mssql container (not the test container container)
+  // 2. /opt/mssql-tools/bin/sqlcmd -S localhost -U SA -P "A_Str0ng_Required_Password"
+  @Override
+  protected void setup(TestDestinationEnv testEnv) throws SQLException {
+    configWithoutDbName = getConfig(db);
+    final String dbName = "db_" + RandomStringUtils.randomAlphabetic(10).toLowerCase();
+
+    final Database database = getDatabase(configWithoutDbName);
+    database.query(ctx -> {
+      ctx.execute("alter database default tablespace users");
+      ctx.execute(
+          "declare c int; begin select count(*) into c from user_tables where upper(table_name) = upper('id_and_name'); if c = 1 then execute immediate 'drop table id_and_name'; end if; end;");
+      ctx.fetch("CREATE TABLE id_and_name(id INTEGER NOT NULL, name VARCHAR(200), born TIMESTAMP WITH TIME ZONE)");
+      ctx.fetch(
+          "INSERT ALL INTO id_and_name (id, name, born) VALUES (1,'picard', TIMESTAMP '2124-03-04 01:01:01') INTO id_and_name (id, name, born) VALUES  (2, 'crusher', TIMESTAMP '2124-03-04 01:01:01') INTO id_and_name (id, name, born) VALUES (3, 'vash', TIMESTAMP '2124-03-04 01:01:01') SELECT 1 FROM DUAL");
```

Comment:
why are we creating this table and inserting rows? the extended test class should handle setting up all test cases. i think this might be copy pasta.

Reply:
oh. i guess the drop table above isn't relative to this test suite either.